### PR TITLE
[FEATURE][processing] New algorithm "Remove duplicates by attribute"

### PR DESCRIPTION
**Pull request by nyalldawson**
Allows for removal of duplicate features, identified using the values in one (or more) field values from the input features.

Optionally any discarded (duplicate) features can be saved to a separate sink.

### CODE REVIEW

File: src/analysis/processing/qgsalgorithmremoveduplicatesbyattribute.cpp [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+}
+
+QStringList QgsRemoveDuplicatesByAttributeAlgorithm::tags() const
+{
+  return QObject::tr( "field,value,same,filter" ).split( ',' );
+}
+
+QString QgsRemoveDuplicatesByAttributeAlgorithm::group() const
+{
+  return QObject::tr( "Vector selection" );
```
Comment (nirvn) [inline src/analysis/processing/qgsalgorithmremoveduplicatesbyattribute.cpp]: A very similar algorithm, delete duplicate geometries, is found in the Vector general group, maybe this new algorithm should also be placed in that group. IMHO it'd be a better fit.

File: src/analysis/processing/qgsalgorithmremoveduplicatesbyattribute.cpp [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+}
+
+void QgsRemoveDuplicatesByAttributeAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterFeatureSource( QStringLiteral( "INPUT" ), QObject::tr( "Input layer" ),
+                QList< int >() << QgsProcessing::TypeVector ) );
+  addParameter( new QgsProcessingParameterField( QStringLiteral( "FIELDS" ), QObject::tr( "Field to match duplicates by" ), QVariant(), QStringLiteral( "INPUT" ), QgsProcessingParameterField::Any, true ) );
+
+  addParameter( new QgsProcessingParameterFeatureSink( QStringLiteral( "OUTPUT" ), QObject::tr( "Filtered (no duplicates)" ) ) );
+  QgsProcessingParameterFeatureSink *failOutput = new QgsProcessingParameterFeatureSink( QStringLiteral( "DUPLICATES" ),  QObject::tr( "Filtered (duplicates)" ),
```
Comment (nirvn) [inline src/analysis/processing/qgsalgorithmremoveduplicatesbyattribute.cpp]: Possible bonus (not blocking this PR in anyway): harmonize the remove duplicate geometries alg's capacity by adding a similar "DUPLICATES" output.

**Review by nirvn — APPROVED** [review]
Looking good.

File: src/analysis/processing/qgsalgorithmremoveduplicatesbyattribute.cpp (hunk 2 above)
Comment (nyalldawson) [reply src/analysis/processing/qgsalgorithmremoveduplicatesbyattribute.cpp]: Ok, I'll do that in a follow up work. I'd like to port that other alg to c++ anyway.

**Pull request merged**